lib/portlayer/exec/commit_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware/vic/pkg/trace"
+)
+
+func TestBatchBlockOnFunc(t *testing.T) {
+	ctx := context.Background()
+
+	operation := func(op trace.Operation) error {
+		time.Sleep(10 * time.Millisecond)
+		return nil
+	}
+
+	// test fetch one request
+	testFetchOne(t, ctx, operation)
+
+	// serialize test: batch size 1, 3 requests
+	// batch size 1 = no batching. Process requests one at a time
+	// total time spent should be within: (# of requests) * (operation wait time)
+	testMultipleBatch(t, ctx, 1, 3, operation)
+
+	// batch size 10, 5 requests
+	// the 5 requests should be processed in one batch
+	// total time spent should be within: 1 * (operation wait time)
```

Comment:
There is variability here - I believe the first operation could always run without any batching, thereby causing the queuing/batching of the remainder. Therefore I'd expect to always have a minimum expected time of `2 * operation wait time`

Reply:
since the incoming requests are sent in tight loop here, when we fetch the first requests from the channel, the rest requests should already been batched. So when we get to the point of calling `operation(members)`, we're doing all operations in one batch. 
But this is still a not general case and only happens if the requests are coming fast enough. So I guess a more general upper bound should be `2 * operation wait time`

Reply:
> To make the tests closer to realistic scenario, I think I could send the requests every certain time interval (for example, 2ms) less than the operation processing time (in this case something larger than 2ms)

The # of batches and the total time spent depend on how fast the requests are coming in - if the time interval between each request is larger than the operation processing time, then it's completely serialized (# of batches = # of requests).